--- origsrc/kmplayer-0.10.0c/configure.in.in	2007-12-21 15:51:44.000000000 -0600
+++ src/kmplayer-0.10.0c/configure.in.in	2008-08-14 03:38:57.145115800 -0500
@@ -58,6 +58,30 @@
     AC_SUBST(LIB_EXPAT)
 fi
 
+KDE_CHECK_HEADER(X11/extensions/Xv.h,
+   have_xv=yes,
+   have_xv=no)
+
+AC_MSG_CHECKING([if kxvplayer can be compiled])
+AC_MSG_RESULT($have_xv)
+
+AC_ARG_WITH(xv,
+    AC_HELP_STRING([--without-xv],[build KMPlayer without XVideo extension [default=with]]),
+    [build_xv=$withval],
+    [build_xv=yes]
+)
+
+if test "$build_xv" != "no"; then
+    if test "$have_xv" != "yes"; then
+        have_xv="no"
+        AC_MSG_WARN([libXv not found on your system; disabling kxvplayer])
+    fi
+else
+    have_xv="no"
+fi
+
+AM_CONDITIONAL(include_kxvplayer, test "$have_xv" = "yes")
+
 KDE_CHECK_HEADER(xine.h,
    have_xine=yes,
    have_xine=no)
--- origsrc/kmplayer-0.10.0c/src/Makefile.am	2007-12-21 15:51:44.000000000 -0600
+++ src/kmplayer-0.10.0c/src/Makefile.am	2008-08-14 03:31:34.191990800 -0500
@@ -32,6 +32,9 @@
 if include_kxineplayer
 kxineplayer_app = kxineplayer
 endif
+if include_kxvplayer
+kxvplayer_app = kxvplayer
+endif
 if include_kgstplayer
 kgstplayer_app = kgstplayer
 endif
@@ -39,7 +42,7 @@
 knpplayer_app = knpplayer
 endif
 
-bin_PROGRAMS= $(kxineplayer_app) kxvplayer $(kgstplayer_app) $(knpplayer_app)
+bin_PROGRAMS= $(kxineplayer_app) $(kxvplayer_app) $(kgstplayer_app) $(knpplayer_app)
 
 noinst_LTLIBRARIES = libkmplayerbackend.la
 libkmplayerbackend_la_SOURCES = kmplayer_backend.skel kmplayer_callback.stub
