--- origsrc/kmplayer-0.10.0c/src/Makefile.am	2007-12-21 15:51:44.000000000 -0600
+++ src/kmplayer-0.10.0c/src/Makefile.am	2008-08-14 03:31:34.191990800 -0500
@@ -5,7 +5,7 @@
 lib_LTLIBRARIES= libkmplayercommon.la
 
 libkmplayercommon_la_SOURCES = viewarea.cpp kmplayerview.cpp playlistview.cpp kmplayercontrolpanel.cpp kmplayerconfig.cpp pref.cpp kmplayerprocess.cpp kmplayer_callback.skel kmplayer_backend.stub kmplayerpartbase.cpp kmplayerplaylist.cpp kmplayer_asx.cpp kmplayer_smil.cpp kmplayer_rp.cpp kmplayer_rss.cpp kmplayer_atom.cpp kmplayer_xspf.cpp triestring.cpp kmplayerpartbase.skel
-libkmplayercommon_la_LDFLAGS = -avoid-version $(all_libraries)
+libkmplayercommon_la_LDFLAGS = -avoid-version -no-undefined $(all_libraries)
 libkmplayercommon_la_LIBADD = -lkmediaplayer $(LIB_KPARTS) $(LIB_KUTILS) $(LIB_EXPAT) -lm $(LIBCAIRO_LIBS) $(LIBQTDBUS)
 
 if include_koffice_support
@@ -16,11 +16,11 @@
 kde_module_LTLIBRARIES= libkmplayerpart.la $(kofficeplugin_lib)
 
 libkmplayerpart_la_SOURCES= kmplayer_part.cpp
-libkmplayerpart_la_LDFLAGS= -avoid-version $(all_libraries) $(KDE_RPATH)
+libkmplayerpart_la_LDFLAGS= -avoid-version $(all_libraries) $(KDE_PLUGIN)
 libkmplayerpart_la_LIBADD= libkmplayercommon.la
 
 libkmplayerkofficepart_la_SOURCES=kmplayer_koffice_part.cpp
-libkmplayerkofficepart_la_LDFLAGS= -avoid-version $(all_libraries) $(KDE_RPATH)
+libkmplayerkofficepart_la_LDFLAGS= -avoid-version $(all_libraries) $(KDE_PLUGIN)
 libkmplayerkofficepart_la_LIBADD= libkmplayercommon.la $(LIB_KOFFICE)
 
 kdeinit_LTLIBRARIES=kmplayer.la
